playtime/init.lua

```diff
@@ -0,0 +1,88 @@
+playtime = {}
+
+local os = os
+local math = math
+local string = string
+
+local current = {}
+
+local key = "playtime:time"
+
+local S = minetest.get_translator("playtime")
+local C = minetest.colorize
+
+--get playtime
+function playtime.get_current_playtime(name)
+	if current[name] then
+		return os.time() - current[name]
+	else
+		return 0
+	end
+end
+
+--get total playtime
+function playtime.get_total_playtime(name)
+	local player = minetest.get_player_by_name(name)
+	if player then
+		return player:get_meta():get_int(key) + playtime.get_current_playtime(name)
+	end
+end
+
+--reset playtime
+function playtime.remove_playtime(name)
+	local player = minetest.get_player_by_name(name)
+	if player then
+		player:get_meta():set_int(key, 0)
+	end
+end
+
+if minetest.is_singleplayer() then
```

Comment:
We don't really need this singleplayer hack. If we did, it didn't belong here, but in an utility mod, as other mods would need it as well. (Something like `minetest_patched.register_on_leaveplayer`).

Reply:
I removed it entirely